app/screens/channel/channel_nav_bar/channel_drawer_button.js

```diff
@@ -74,62 +77,89 @@ class ChannelDrawerButton extends PureComponent {
         this.setState({opacity: value > 0 ? 0.1 : 1});
     };
 
-    handlePress = () => {
-        EventEmitter.emit('open_channel_drawer');
-    };
+    handlePress = wrapWithPreventDoubleTap(() => {
+        this.props.openDrawer();
+    });
 
     render() {
         const {
             currentTeamId,
             mentionCount,
             messageCount,
             myTeamMembers,
-            theme
+            theme,
+            type
         } = this.props;
         const style = getStyleFromTheme(theme);
 
-        let mentions = mentionCount;
-        let messages = messageCount;
-
-        const members = Object.values(myTeamMembers).filter((m) => m.team_id !== currentTeamId);
-        members.forEach((m) => {
-            mentions = mentions + (m.mention_count || 0);
-            messages = messages + (m.msg_count || 0);
-        });
-
-        let badgeCount = 0;
-        if (mentions) {
-            badgeCount = mentions;
-        } else if (messages) {
-            badgeCount = -1;
+        let badge;
+        if (type === 'left') {
```

Comment:
Why not just use a separate component at this point? We're changing out almost the entire component depending on the `type` field?

Reply:
We can